src/main/scala/cromwell/engine/backend/jes/JesBackend.scala

```diff
@@ -152,10 +152,29 @@ class JesBackend extends Backend with LazyLogging with ProductionJesAuthenticati
   }
 
   /**
-   * Delete authentication file in gcs once workflow is in a terminal state.
+   * Delete authentication file in GCS once workflow is in a terminal state.
+   *
+   * First queries for the existence of the auth file, then deletes it if it exists.
+   * If either of these operations fails, then a Future.failure is returned
    */
   override def cleanUpForWorkflow(workflow: WorkflowDescriptor)(implicit ec: ExecutionContext): Future[Unit] = authenticated { connection =>
-    Future(connection.storage.deleteObject(GoogleCloudStoragePath(gcsAuthFilePath(workflow))))
+    val gcsAuthFile = GoogleCloudStoragePath(gcsAuthFilePath(workflow))
+    def gcsCheckAuthFileExists(prior: Option[Boolean]): Boolean = connection.storage.exists(gcsAuthFile)
+    def gcsAttemptToDeleteObject(prior: Option[Unit]): Unit = connection.storage.deleteObject(gcsAuthFile)
+    withRetry(gcsCheckAuthFileExists, s"Workflow ${workflow.shortId} - failed to query for auth file: $gcsAuthFile") match {
```

Comment:
There are a whole bunch of these "Workflow ${workflow.shortID} - blahblah" strings. Perhaps a function which hardwires the previous bit and takes another string to append?